channeld/channeld.c

```diff
@@ -5228,15 +5228,15 @@ static void handle_funding_depth(struct peer *peer, const u8 *msg)
 			status_debug("handle_funding_depth: Setting short_channel_ids[LOCAL] to %s",
 				type_to_string(tmpctx,
 					       struct short_channel_id,
-					       (scid ? scid : &peer->local_alias)));
+					       (scid ? scid : peer->local_alias)));
 			/* If we know an actual short_channel_id prefer to use
 			 * that, otherwise fill in the alias. From channeld's
 			 * point of view switching from zeroconf to an actual
 			 * funding scid is just a reorg. */
 			if (scid)
 				peer->short_channel_ids[LOCAL] = *scid;
 			else
-				peer->short_channel_ids[LOCAL] = peer->local_alias;
+				peer->short_channel_ids[LOCAL] = *peer->local_alias;
```

Comment:
This will crash if we have an unconfirmed channel that predates the introduction of the aliases. This might be the right place to use a dummy scid for this purpose, or we make the array an array of pointers which would then allow us to just copy the reference here rather than dereferencing.

Reply:
ups! maybe we can add a dummy value and log in to debug mode that the local_alias is null?